extensions/miro/src/create-board.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useState } from "react";
+import { Action, ActionPanel, Detail, Form, showToast, Toast, useNavigation } from "@raycast/api";
+import * as miro from "./oauth/miro";
+import ListBoards from "./list-boards";
+
+interface CreateBoardProps {
+  name: string;
+  description: string;
+}
+
+export default function CreateBoard() {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [nameError, setNameError] = useState<string | undefined>();
+
+  const { push } = useNavigation();
+
+  useEffect(() => {
+    (async () => {
+      try {
+        await miro.authorize();
+        setIsLoading(false);
+      } catch (error) {
+        console.error(error);
+        setIsLoading(false);
+        await showToast({ style: Toast.Style.Failure, title: String(error) });
+      }
+    })();
+  }, []);
+
+  function dropNameErrorIfNeeded() {
+    if (nameError && nameError.length > 0) {
+      setNameError(undefined);
+    }
+  }
+
+  if (isLoading) {
+    return <Detail isLoading={isLoading} />;
+  }
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm
+            title="Create Board"
+            onSubmit={async (values: CreateBoardProps) => {
+              try {
+                await miro.createItem(values.name, values.description);
+                await showToast({
+                  style: Toast.Style.Success,
+                  title: "Board created",
+                });
+                push(<ListBoards />);
```

Comment:
I don't know if you should push the list here. If the user presses escape, they will come back to the form. 

Instead you might want to render the list at the top level when it's done?

Reply:
I put there the new `launchCommand` API.